src/App/Application.cpp

```diff
@@ -2773,16 +2773,23 @@ void Application::ExtractUserPath()
 #if defined(FC_OS_LINUX) || defined(FC_OS_CYGWIN) || defined(FC_OS_BSD)
     // Default paths for the user specific stuff
     struct passwd *pwd = getpwuid(getuid());
-    if (pwd == NULL)
-        throw Base::RuntimeError("Getting HOME path from system failed!");
-    mConfig["UserHomePath"] = pwd->pw_dir;
+
     if (!userHome.isEmpty()) {
         mConfig["UserHomePath"] = userHome.toUtf8().data();
+    } else if (pwd != NULL) {

```

Comment:
As long as you're touching the code anyway, I think we should get these old `NULL`s changed to `nullptr`.